src/rules/__tests__/normalizeSyntheticEvent.test.js

```diff
@@ -111,4 +111,70 @@ describe('normalizeSyntheticEvent', function () {
       }
     });
   });
+
+  it(
+    'does not override meta of first normalized event if synthetic event is ' +
+      'normalized multiple times',
+    function () {
+      var syntheticEvent = {};
+      var syntheticEventMeta1 = { $rule: { id: 'ABC123' } };
+      var syntheticEventMeta2 = { $rule: { id: 'ABC456' } };
+      var normalizedSyntheticEvent1 = normalizeSyntheticEvent(
+        syntheticEventMeta1,
+        syntheticEvent
+      );
+      var normalizedSyntheticEvent2 = normalizeSyntheticEvent(
+        syntheticEventMeta2,
+        syntheticEvent
+      );
+
+      expect(normalizedSyntheticEvent1.$rule.id).toEqual('ABC123');
+      expect(normalizedSyntheticEvent2.$rule.id).toEqual('ABC456');
+    }
+  );
+
+  it('leaves syntheticEvent objects alone that are not plain objects', function () {
+    // source: https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent/CustomEvent
+    function CustomEventShim(event, params) {
+      params = params || {
+        bubbles: false,
+        cancelable: false,
+        detail: undefined
+      };
+      var evt = document.createEvent('CustomEvent');
+      evt.initCustomEvent(
+        event,
+        params.bubbles,
+        params.cancelable,
+        params.detail
+      );
+      return evt;
+    }
+
+    if (typeof window.CustomEvent !== 'function') {
+      CustomEventShim.prototype = window.Event.prototype;
+      window.CustomEvent = CustomEventShim;
+    }
+
+    var syntheticEvent = new window.CustomEvent('swipe', {
+      swipeAngle: 90
+    });
+    var oldSyntheticEventReference = syntheticEvent;
+    var syntheticEventMeta1 = { $rule: { id: 'ABC123' } };
+    var syntheticEventMeta2 = { $rule: { id: 'ABC456' } };
+    var normalizedSyntheticEvent1 = normalizeSyntheticEvent(
+      syntheticEventMeta1,
+      syntheticEvent
+    );
+    var normalizedSyntheticEvent2 = normalizeSyntheticEvent(
+      syntheticEventMeta2,
+      syntheticEvent
+    );
+
+    expect(oldSyntheticEventReference === syntheticEvent).toBeTrue();
```

Comment:
I don't think this assertion is providing any value, since this will always be true regardless of what happens inside Turbine. Turbine has no way to change the reference of the `syntheticEvent` or `oldSyntheticEventReference` variables. It can modify the object's content, but not the references to that object.

Reply:
I compared the wrong things, great catch. I was trying to assert that what's handed back from the function call is the same thing.